### ISSUE TITLE
Some ambiguous or misleading configuration names and descriptions

### ISSUE THREAD

**Issue by danemacmillan** [algolia/algoliasearch-magento-2#508]
Versions:
- Magento: `2.2.4`
- Algolia: `1.7.0`

I've started using this extension from Algolia and testing it out. It's really nice so far. However, I found a few of the configurations in the Magento2 admin are either ambiguous or misleading. I think they could benefit from some clarity.

### Credentials and Setup > Enable Search
> If set to No, search will be done by Algolia but renderered by Magento 

This is false. Setting this `Yes/No` option to `No` will in fact completely disable Algolia search. It won't do as it describes. In reading its description, it leads me to think that the search results will come from Algolia through the backend and leave Magento2 to render the results using its standard templates. In other words, instead of seeing the search results as an "Instant Search Results Page," it would just be Magento2 with the data coming from Algolia in the backend (no asynchronous request) instead of Magento2's fulltextsearch. This is verifiable by first 1) comparing search results with this setting toggled to `Yes` and then `No`, and noting the obvious difference in the results; and secondly, 2) by reading the source code itself at https://github.com/algolia/algoliasearch-magento-2/blob/master/Adapter/Algolia.php#L113 where `isAllowed` calls `isEnabledFrontEnd`, which has the following body (https://github.com/algolia/algoliasearch-magento-2/blob/master/Helper/ConfigHelper.php#L236):

```
    public function isEnabledFrontEnd($storeId = null)
    {
        // Frontend = Backend + Frontend
        return (bool) $this->configInterface->getValue(self::ENABLE_BACKEND, ScopeInterface::SCOPE_STORE, $storeId)
            && (bool) $this->configInterface->getValue(self::ENABLE_FRONTEND, ScopeInterface::SCOPE_STORE, $storeId);
    }
```

Besides this, the actual name of the underlying config is entirely different from the words used to describe it both briefly and at length. "Enable Search" is technically `enable_frontend`.

This is confusing.

### Credentials and Setup > Enable Instant Search Results Page
> If set to Yes, the products inside the results pages will be searchable the refined results updated as-you-type. It requires your theme to expose a top.search and content block. 

This one actually seems to do what the previous configuration purportedly does. The language in the description leads me to think, though, that this setting will only remove the filter/search bar at the top of the results page, which provides instant filtering when typing into it, but leave the other "instant" or asynchronous parts of the results page, including pagination (or infinite scroll) and attribute/category filtering.

### Advanced > Make SEO request
> Choose if a query to Algolia is done by the backend for SEO when instant search is enabled 

I have no idea what this does. In looking at the code at https://github.com/algolia/algoliasearch-magento-2/blob/master/Adapter/Algolia.php#L122 I can see where the condition for its use is active, but it doesn't actually reveal much.

What does this *actually* do? Nowhere is this elaborated upon in the documentation.

I've compared results with this toggled to `Yes` and `No`, and cannot see what affect this is having.

**Comment by JanPetr** [comment]
Hello @danemacmillan,

thank you for the great suggestions! We are aware we're lacking clarity in some configuration directives. In the next version we'll work on that make sure we incorporate what you've suggested here.

Thanks again for your help 🙌

**Comment by JanPetr** [comment]
The mentioned issues are solved in #516. Would you mind taking a look @danemacmillan and share your feedback? :)

**Comment by damcou** [comment]
Hi @danemacmillan , As @JanPetr told you, your suggestions helped us a lot for our next release. 

I'll close this ticket but feel free to reopen it if you have any feedback ;).

Thanks again for your help !

Best.

Damien
